docs/rules/jsx-max-depth.md

```diff
@@ -0,0 +1,78 @@
+# Validate JSX maximum depth (react/jsx-max-depth)
+
+This option validates a specific depth for JSX.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```jsx
+<App>
+  <Foo>
+    <Bar>
+      <Baz />
+    </Bar>
+  </Foo>
+</App>
+
+```
+
+## Rule Options
+
+It takes an option as the second parameter which can be a positive number for depth count.
+
+```js
+...
+"react/jsx-no-depth": [<enabled>, <number>]
```

Comment:
let's make an initial schema that's extensible; in other words, `{ "max": <number> }` instead of just the number.